generic/tests/trans_hugepage_relocated.py

```diff
@@ -63,47 +68,30 @@ def nr_hugepage_check(sleep_time, wait_time):
     s_time = int(re.findall(r"scan_sleep_millisecs:(\d+)", thp_cfg)[0]) / 1000
     w_time = int(re.findall(r"alloc_sleep_millisecs:(\d+)", thp_cfg)[0]) / 1000
 
-    try:
-        test.log.info("Turn off swap in guest")
-        s, o = session.cmd_status_output("swapoff -a")
-        if s != 0:
-            test.log.warning("Didn't turn off swap in guest")
-        s, o = session.cmd_status_output("cat /proc/meminfo")
-        mem_free_filter = r"MemFree:\s+(.\d+)\s+(\w+)"
-        guest_mem_free, guest_unit = re.findall(mem_free_filter, o)[0]
-        if re.findall("[kK]", guest_unit):
-            guest_mem_free = str(int(guest_mem_free) / 1024)
-        elif re.findall("[gG]", guest_unit):
-            guest_mem_free = str(int(guest_mem_free) * 1024)
-        elif re.findall("[mM]", guest_unit):
-            pass
-        else:
-            guest_mem_free = str(int(guest_mem_free) / 1024 / 1024)
+    logging.info("Turn off swap in guest")
+    s, o = session.cmd_status_output("swapoff -a")
+    if s != 0:
+        logging.warning("Didn't turn off swap in guest")
+    s, o = session.cmd_status_output("cat /proc/meminfo")
+    mem_free_filter = r"MemFree:\s+(.\d+)\s+(\w+)"
+    guest_mem_free, guest_unit = re.findall(mem_free_filter, o)[0]
+    if re.findall("[kK]", guest_unit):
+        guest_mem_free = int(guest_mem_free) / 1024
+    elif re.findall("[gG]", guest_unit):
+        guest_mem_free = int(guest_mem_free) * 1024
+    elif re.findall("[mM]", guest_unit):
+        pass
+    else:
+        guest_mem_free = int(guest_mem_free) / 1024 / 1024
 
-        file_size = min(1024, int(guest_mem_free) / 2)
-        cmd = "mount -t tmpfs -o size=%sM none /mnt" % file_size
-        s, o = session.cmd_status_output(cmd)
-        if nr_hugetlbfs:
-            hugepage_cfg = open(hugetlbfs_path, "w")
+    file_size = min(1024, int(guest_mem_free / 2))
+    cmd = "mount -t tmpfs -o size=%sM none /mnt" % file_size
+    s, o = session.cmd_status_output(cmd)
+    if nr_hugetlbfs:
+        with open(hugetlbfs_path, "w") as hugepage_cfg:
             hugepage_cfg.write(str(nr_hugetlbfs))
-            hugepage_cfg.close()
-
-        if not os.path.isdir('/space'):
-            os.makedirs('/space')
-        if os.system("mount -t tmpfs -o size=%sM none /space" % vmsm):
-            test.error("Can not mount tmpfs")
 
-        # Try to make some fragment in memory
-        # The total size of fragments is vmsm
-        count = vmsm * 1024 / 4
-        cmd = "for i in `seq %s`; do dd if=/dev/urandom of=/space/$i" % count
-        cmd += " bs=4K count=1 & done"
-        test.log.info("Start to make fragment in host")
-        s = process.system(cmd, verbose=False, shell=True)
-        if s != 0:
-            test.error("Can not dd in host")
-    finally:
-        process.run("umount /space", verbose=False, shell=True)
+    thp_fragment_tool.execute_tool(test)
```

Comment:
Suggest adding some log info when start execute the tool .

Reply:
Included !